src/Node_verify.cpp

```diff
@@ -354,6 +354,212 @@ void Node::verify_vdf(std::shared_ptr<const ProofOfTime> proof, const uint32_t c
+	static int prev_num_cpu = 0;
+	static int prev_num_ocl = 0;
+
+	const auto& segments = proof->segments;
+	bool is_valid = !segments.empty();
+	size_t invalid_segment = -1;
+	size_t num_invalid_segment = 0;
+
+	const int num_chain = (verify_vdf_rewards && proof->reward_addr) ? 3 : 2;
+	const int num_point = segments.size() * num_chain;
+	if(!num_point) {
+		throw std::logic_error("no segments");
+	}
+
+	std::vector<hash_t> point;
+	std::vector<hash_t> point_verify;
+	std::vector<uint32_t> num_iters;
+	uint32_t max_iters = 0;
+
+	point.resize(num_point);
+	point_verify.resize(num_point);
+	num_iters.resize(num_point);
+
+	// comment: copy/infuse points to be hashed, points to verify against, number of iters, 3x vectors
+	for(int chain = 0; chain < num_chain; ++chain)
+	{
+		for(size_t i = 0, j = chain * segments.size(); i < segments.size(); ++i, ++j)
+		{
+			if(i > 0) {
+				point[j] = chain < 2 ? segments[i-1].output[chain] : proof->reward_segments[i-1];
+			} else {
+				point[j] = proof->input[chain % 2];
+				if(chain < 2) {
+					if(auto infuse = proof->infuse[chain]) {
+						point[j] = hash_t(point[j] + (*infuse));
+					}
+				} else if(proof->reward_addr) {
+					point[j] = hash_t(point[j] + (*proof->reward_addr));
+				}
+			}
+			point_verify[j] = chain < 2 ? segments[i].output[chain] : proof->reward_segments[i];
+			num_iters[j] = segments[i].num_iters;
+			max_iters = std::max(max_iters, num_iters[j]);
+		}
+	}
+
+	// comment: find load balance between cpu/opencl, based on last block, or 50/50 if none
+	// comment: calc time per point for cpu/opencl, estimate point balance to get equal total time
+	// comment: limit minimum usage to 10% of points to either cpu/opencl
+	int num_point_cpu = num_point / 2;
+	int num_point_ocl = num_point - num_point_cpu;
+	if(prev_time_cpu > 0 && prev_time_ocl > 0 && prev_num_cpu > 0 && prev_num_ocl > 0) {
+		const int prev_num_point = prev_num_cpu + prev_num_ocl;
+		const uint64_t cpu_micro_pt = prev_time_cpu / prev_num_cpu;
+		const uint64_t ocl_micro_pt = prev_time_ocl / prev_num_ocl;
+		uint64_t balance_pct = (((prev_num_point * ocl_micro_pt) / (cpu_micro_pt + ocl_micro_pt)) * 100) / prev_num_point;
+
+		// TODO: find out real practical lower limit on each side of cpu/openCL, percent or points
+		balance_pct = std::clamp(balance_pct, (uint64_t)10, (uint64_t)90);
+
+		num_point_cpu = (num_point * balance_pct) / 100;
+		num_point_ocl = num_point - num_point_cpu;
+	}
+
+	// comment: find balanced thread number based on recommended hardware concurrency
+	// comment: - too few threads will choke cpu calc potential
+	// comment: - too many threads will choke opencl thread potential
+	// comment: - hardware_concurrency() looks to give logical cpu units, not physical
+	// comment: balance found, good middleground: 'logic cpu units' + 50%
+	// comment: limit minimum to 8, maximum to 72
+	const uint32_t async_num = std::clamp((std::thread::hardware_concurrency() * 15) / 10, (uint32_t)8, (uint32_t)72);
```

Comment:
`std::clamp<uint32_t>(...)` would look better

Reply:
Will fix, depending on new variant of load balance.